app/src/main/java/me/phh/treble/app/Lid.kt

```diff
@@ -26,6 +26,35 @@ object Lid: EntryStartup {
         PowerManager::class.java.getMethod("goToSleep", Long::class.java).invoke(powerManager, SystemClock.uptimeMillis())
     }
 
+    fun huawei(ctxt: Context) {
+        val sensorManager = ctxt.getSystemService(SensorManager::class.java)
+        // Huawei Hall sensor for magnetic cover : ID = 65538
+        val sensors = listOf("HALL sensor")
+        val lidSensor = sensorManager.getSensorList(Sensor.TYPE_ALL).firstOrNull() { sensor -> sensors.any { name -> sensor.name.contains(name)}}
+        if(lidSensor == null) {
+            Log.d("PHH", "Failed finding sensor for lid wakeup")
+            for(s in sensorManager.getSensorList(Sensor.TYPE_ALL)) {
+                Log.d("PHH", " - '${s.name}'")
+            }
+        }
```

Comment:
Well please stop here (just a simple `return`) if you can't find a sensor, otherwise TrebleApp will crash